crates/bevy_ecs/macros/src/component.rs

```diff
@@ -213,6 +223,7 @@ fn parse_component_attr(ast: &DeriveInput) -> Result<Attrs> {
         on_replace: None,
         on_remove: None,
         requires: None,
+        immutable: false,
```

Comment:
Reading this reminds me of the double negative problem with naming things. Typically we want to use "positive" variable names. Maybe we could invert this and name the field `mutable`?

Reply:
My rationale here was `false` is typically what people think of when they think of a default value for a `bool`. I do agree that it could just as easily be `mutable` though.

Reply:
Personally, `mutable` sounds better because I think that with this PR, Mutable is *adding* functionality, which is a positive action to me.

In the general case, I also prefer `enabled` over `disabled` for similar reasons.

It's a minor detail, and consistency is more important. This is mostly a nit.

Reply:
No I agree these are good points. I guess my perspective is a little backwards because we're _adding_ immutable components, even though it's really refactoring `Component` to make mutability not implicit (well, except outside of the derive macro, since it'd be annoying to annotate every component with `#[component(mutable)]`)

Reply:
I am fine to keep it as `#[component(immutable)]` or to change it to `#[component(mutable = false)]`. Personally I prefer the former, but I don't care strongly. We should not have to label mutable components with anything, it should be the default.